twopaneview/src/TwoPaneView.tsx

```diff
@@ -62,85 +59,63 @@ export class TwoPaneView extends Component<Props, State> {
-          return this.renderPane1(this.getEntireSize());
-        }
-        else {
-          return this.renderPane2(this.getEntireSize());
-        }
+    const paneMode = this.getPaneMode();
+    if (paneMode === PaneMode.Auto) {
+      // TODO:  add logic for auto-detecting width > threshold
+      if (this.state.spanning) {
+        return this.renderBothPanes();
       }
-      return this.renderBothPanes();
+      return this.renderPaneWithPriority();
     }
-    if (this.state.paneMode === PaneMode.Dual) {
-      return this.renderDualPanes();
+    if (paneMode === PaneMode.Single) {
+      return this.renderPaneWithPriority();
     }
-    if (this.state.panePriority === PanePriority.Pane1) {
-      return this.renderPane1(this.getEntireSize());
+    if (paneMode === PaneMode.Double) {
+      return this.renderBothPanes();
     }
-    return this.renderPane2(this.getEntireSize());
   }
 
-  renderDualPanes() {
-    const children = React.Children.toArray(this.props.children);
-
-    const items = [];
-    if (children.length > 0) {
-      items.push(this.renderPane1(this.getLeftSize()));
+  renderPaneWithPriority() {
+    if (this.getPanePriority() === PanePriority.Pane1) {
+      return this.renderPane1();
     }
-
-
-    if (children.length > 1) {
-      items.push(this.renderPane2(this.getRightSize()));
+    else {
+      return this.renderPane2();
     }
-
-    return items;
   }
-
+  
   renderBothPanes() {
-    let horizontal = this.state.dims.width >= this.state.dims.height;
     const children = React.Children.toArray(this.props.children);
 
     const items = [];
-    if (children.length > 0) {
-      items.push(this.renderPane1(horizontal ? this.getLeftSize() : this.getTopSize()));
-    }
-
+    items.push(this.renderPane1());
     items.push(this.renderSeparator());
-
-    if (children.length > 1) {
-      items.push(this.renderPane2(horizontal ? this.getRightSize() : this.getBottomSize()));
-    }
+    items.push(this.renderPane2());
 
     return items;
   }
 
-  renderPane1(size: any) {
+  renderPane1() {
     const children = React.Children.toArray(this.props.children);
-    if (children.length > 0) {
-      return (
-        <View key={PanePriority.Pane1} style={{width:size.width, height:size.height}}>
-          {children[0]}
-        </View>
-      );
-    }
+    return (
+      <View key={PanePriority.Pane1} style={{flex: 1}}>
+        {children.length > 0 ? children[0] : null}
+      </View>
+    );
   }
 
-  renderPane2(size: any) {
+  renderPane2() {
     const children = React.Children.toArray(this.props.children);
-    if (children.length > 1) {
-      return (
-        <View key={PanePriority.Pane2} style={{width:size.width, height:size.height}}>
-          {children[1]}
-        </View>
-      );
-    }
+    return (
+      <View key={PanePriority.Pane2} style={{flex: 1}}>
+        {children.length > 1 ? children[1] : null}
+      </View>
+    );
   }
 
   renderSeparator() {
```

Comment:
My advise here as previously use Hinge component from dualscreeninfo

Reply:
I am tracking this with issue #28.  Hinge currently lives in dualscreeninfo, and also needs some work (hard-coded hinge width, can't yet handle vertical orientation).  Will tackle this later.

---
In reply to: [426024497](https://github.com/microsoft/react-native-dualscreen/pull/34#discussion_r426024497) [](ancestors = 426024497)